Shocksupershaman - Copy/src/main/java/com/example/shocksupershaman/shocktask1/service/UserService.java

```diff
@@ -10,26 +11,40 @@
 
 @Service
 public class UserService {
+
     private final Map<String, User> db = new HashMap<>();
+    private final UserMapper userMapper;
+
+    public UserService(UserMapper userMapper) {
+        this.userMapper = userMapper;
+    }
 
     public List<UserDTO> getAllUsers() {
         return db.values().stream()
-                .map(User::toDTO)
+                .map(userMapper::toDTO)
                 .collect(Collectors.toList());
     }
 
-    public UserDTO getUserById(String id) throws UserNotFoundException {
+    public UserDTO getUserById(String id) {
+
         User user = db.get(id);
         if (user == null) {
             throw new UserNotFoundException();
         }
-        return user.toDTO();
+        return userMapper.toDTO(user);
     }
 
     public User createUser(User user) {
+

```

Comment:
Remove this check. Spring is validating request body internally if you defined rules for fields suck as NotNull